### PR TITLE
Break MCP's dependency on Galley metadata

### PR DESCRIPTION
**Pull request by ozevren**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pkg/mcp/snapshot/inmemory.go [hunk 1]
```diff
@@ -146,6 +146,16 @@ func (s *InMemory) Version(collection string) string {
 	return s.versions[collection]
 }
 
+// Collections is an implementation of Snapshot.Collections
+func (s *InMemory) Collections() []string {
+	result := make([]string, 0, len(s.resources))
+	for col := range s.resources {
+		result = append(result, col)
+	}
+
```
Comment (ayj) [inline pkg/mcp/snapshot/inmemory.go]: nit: inconsistent extra space here vs. the changes in galley/pkg/config/processing/snapshotter/snapshot.go.

File: pkg/mcp/snapshot/snapshot.go [hunk 2]
```diff
@@ -107,7 +108,8 @@ func (si *StatusInfo) LastWatchRequestTime() time.Time {
 }
 
 // Watch returns a watch for an MCP request.
-func (c *Cache) Watch(request *source.Request, pushResponse source.PushResponseFunc, peerAddr string) source.CancelWatchFunc { // nolint: lll
+func (c *Cache) Watch(
```
Comment (ayj) [inline pkg/mcp/snapshot/snapshot.go]: Might as well go all the way and wrap after each parameter and delete the `nolint: lll`. e.g.

```golang
func (c *Cache) Watch(
	request *source.Request, 
    pushResponse source.PushResponseFunc, 
    peerAddr string
) source.CancelWatchFunc { 
    group := g.groupIndex(...)
```

File: pkg/mcp/snapshot/snapshot_test.go [hunk 3]
```diff
@@ -39,6 +39,14 @@ type fakeSnapshot struct {
 
 func (fs *fakeSnapshot) Resources(collection string) []*mcp.Resource { return fs.resources[collection] }
 func (fs *fakeSnapshot) Version(collection string) string            { return fs.versions[collection] }
+func (fs *fakeSnapshot) Collections() []string {
+	result := make([]string, 0, len(fs.resources))
+	for collection := range fs.resources {
```
Comment (geeknoid) [inline pkg/mcp/snapshot/snapshot_test.go]: Can you just replace this loop with result = append(result, fs.resources...)?

File: pkg/mcp/snapshot/snapshot.go [hunk 4]
```diff
@@ -153,7 +155,8 @@ func (c *Cache) Watch(request *source.Request, pushResponse source.PushResponseF
 	cancel := func() {
 		c.mu.Lock()
 		defer c.mu.Unlock()
-		if info, ok := c.status[group]; ok {
+		var ok bool
+		if info, ok = c.status[group]; ok {
```
Comment (ayj) [inline pkg/mcp/snapshot/snapshot.go]: It seems a little surprising to see a variable from another calling context re-used here. Did this fix a linter/compiler warning?

File: pkg/mcp/snapshot/snapshot.go [hunk 5]
```diff
@@ -298,32 +301,32 @@ func (c *Cache) GetGroups() []string {
...
 	if snapshot, ok := c.snapshots[group]; ok {
 
-		snapshots := make([]Info, 0, len(metadata.Types.All()))
-		collections := make([]string, 0, len(metadata.Types.All()))
+		var snapshots []Info
+		collections := make([]string, 0, len(c.snapshots))
 
-		for _, info := range metadata.Types.All() {
-			collections = append(collections, info.Collection.String())
+		for _, col := range snapshot.Collections() {
```
Comment (geeknoid) [inline pkg/mcp/snapshot/snapshot.go]: Would collections = append(collection, snapshot.Collections()...) work here?

File: pkg/mcp/snapshot/snapshot_test.go (hunk 3 above)
Comment (ozevren) [reply pkg/mcp/snapshot/snapshot_test.go]: It won't work. fs.resources is a map.

File: pkg/mcp/snapshot/snapshot.go (hunk 4 above)
Comment (ozevren) [reply pkg/mcp/snapshot/snapshot.go]: yes. It was a golang warning. Reverting. Good catch.

File: pkg/mcp/snapshot/snapshot.go (hunk 5 above)
Comment (ozevren) [reply pkg/mcp/snapshot/snapshot.go]: Yup. Fixed. Thanks.

**Pull request merged**